modules/ui/sections/raw_membership_editor.js

```diff
@@ -114,7 +116,31 @@ export function uiSectionRawMembershipEditor(context) {
             membership.role = roles.length === 1 ? roles[0] : roles;
         });
 
-        return memberships;
+        const existingRelations = memberships
+            .filter(membership => !recentlyAdded.has(membership.relation.id))
+            .map(membership => ({
+                ...membership,
+                // We only sort relations that were not added just now.
+                // Sorting uses the same label as shown in the UI.
+                // If the label is not unique, the relation ID ensures
+                // that the sort order is still stable.
+                _sortKey: [
+                    baseDisplayValue(membership.relation),
+                    membership.relation.id,
+                ].join('-'),
+            }))
+            .sort((a, b) => a._sortKey.localeCompare(b._sortKey));
```

Comment:
Can we use `Intl.Collator`, setting options that include [numeric sorting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#numeric)? `String.localeCompare` does a lexicographical comparison, which is normally fine, but many of these labels will be numeric or contain numbers, particularly route relations, which don’t make as much sense when sorted alphabetically. It’s worse in OpenHistoricalMap’s fork, which appends a date range to every element’s label: OpenHistoricalMap/iD#184.

Reply:
done, using `String#localeCompare`'s [third argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#options), rather than directly invoking `Intl.Collator`, to avoiding writing a fallback [like we've done for other `Intl` APIs](https://github.com/openstreetmap/iD/blob/b2b397c7ce2325e9f770f586d7da811681723a21/modules/core/localizer.js#L229)